base/linalg/lq.jl

```diff
@@ -88,22 +88,25 @@ end
 
 size(A::LQPackedQ) = size(A.factors)
 
-full(A::LQ) = A[:L]*A[:Q]
-#=
-We construct the full eye here, even though it seems ineffecient, because
-every element in the output matrix is a function of all the elements of
-the input matrix. The eye is modified by the elementary reflectors held
-in A, so this is not just an indexing operation. Note that in general
-explicitly constructing Q, rather than using the ldiv or mult methods,
-may be a wasteful allocation.
-=#
-function full{T}(A::LQPackedQ{T}; thin::Bool=true)
+convert(::Type{Matrix}, A::LQ) = convert(Array, A[:L]*A[:Q])
+convert(::Type{Array}, A::LQ) = convert(Matrix, A)
+full(A::LQ) = convert(Array, A)
+
+function convert{T}(::Type{Matrix}, A::LQPackedQ{T}; thin::Bool = true)
```

Comment:
If I understood the above correctly, then the approach in this PR and #17079 is in keeping with your thoughts, yes? #17079 now replaces the few `convert(Array, X; thin = false)` calls in `base/` (and several in `test/`) with calls to an unexported function `thickQ`, allowing #17082 to deprecate `full` altogether (though I haven't updated the latter yet). Thanks again! I hope JuliaCon went well and you now have opportunity for a well-earned breather!